### ISSUE TITLE
Python Interface

### ISSUE THREAD

**Issue by tvandera** [ExaScience/smurff#7]
- Python interface of "ExaScience/smurff" be lifted to functionality provided in "jaak-s/macau"
- See also integration test cases
- Existing tests in "jaak-s/macau"
   - https://github.com/jaak-s/macau/blob/master/python/macau/test_macau.py
   - All tests should be transferred to "ExaScience/smurff"
   - https://github.com/ExaScience/smurff/blob/smurff/python/smurff/test/test_bpmf.py

**Comment by tvandera** [comment]
All cmd-line options should available via python interface

**Comment by ipasechnikov** [comment]
Currently we are working on python interface in `smurff-python` branch. For now there is a reduced amount of command line arguments available in python `smurff` function that does not return any result.

So here is the plan:

1. Update `.pxd` files. Remove extra `Session` files and add new `ISession.pxd`
2. Use result handling from old version of `smurff.pyx`
3. Use `step()` logic from old version of `smurff.pyx`
4. Add remaining command line arguments so that to enable full control of smurff from python

**Comment by ipasechnikov** [comment]
Python interface is almost done. You can check it out in `smurff-python` branch, commit 828ad1fdbae09160044d92836bef848ba785b2a8.
1. Updated list of `.pxd` files. `Session.pxd` is removed and new `ISession.pxd` is added
2. `smurff` function now returns a `MatrixConfig` from `ISession::getResult` and converts it into [`scipy.sparse.coo_matrix`](https://docs.scipy.org/doc/scipy-0.19.1/reference/generated/scipy.sparse.coo_matrix.html). Also I suppose that we can add a condition to create a `numpy.ndarray` if `MatrixConfig` is a dense matrix.
3. `ISession::run` is still used instead of `ISession::step` cause to start stepping you need to init session using `Session::init` method that `ISession` does not provide. And I still don't quite get it why we want to step instead of just running session. If we really need that step, then we can consider updating `ISession` to provide `init` method or some other solution
4. Most command line arguments are now available from python. 2 arguments for setting noise model actually do nothing. `precision` and `adaptive` arguments should set `NoiseModel` but currently only `CmdSession` can set noise model because it has special lambda function [`set_noise_model`](https://github.com/ExaScience/smurff/blob/828ad1fdbae09160044d92836bef848ba785b2a8/lib/smurff-cpp/SmurffCpp/Sessions/CmdSession.cpp#L35). We should definitely fix this

Feel free to suggest any improvements or ideas.

**Comment by ipasechnikov** [comment]
I suppose we may consider Python module finished.

`smurff-python` branch is merged into `master` branch and removed. So all Python module stuff is now in `master` branch.

Latest for now commit d80241277a7b6ec2530678158b5cf20f399fca40 has an updated `README.md` that contains a [small section](https://github.com/ExaScience/smurff#python-wrapper-installation-on-ubuntu) describing how to build and install Python module.

There are only a few questions left before we can close this issue:
1. What should we do with old [`setup.py`](https://github.com/ExaScience/smurff/blob/master/setup.py) and [`setup_orig.py`](https://github.com/ExaScience/smurff/blob/master/setup_orig.py) files? Should we just remove them cause we don't need them anymore?
2. Do we need [`requirements.txt`](https://github.com/ExaScience/smurff/blob/master/requirements.txt) file? It looks like an auto-generated `requires.txt` file from `egg-info` directory

**Comment by tvandera** [comment]
- I think we can remove setup.py and setup_orig.py, and requirements.txt.

**Comment by ipasechnikov** [comment]
And one last thing. We don't need [test.py](https://github.com/ExaScience/smurff/blob/master/test.py) file, do we? We can use [smurff-mm.py](https://github.com/ExaScience/smurff/blob/master/python/smurff/examples/smurff-mm.py) instead. But this script uses  [chembl-IC50-346targets.mm](https://github.com/ExaScience/smurff/blob/master/python/smurff/examples/chembl-IC50-346targets.mm) and [chembl-IC50-test.mm](https://github.com/ExaScience/smurff/blob/master/python/smurff/examples/chembl-IC50-test.mm) that I added to repository. Is it fine to keep them here or I should remove them?

**Comment by tvandera** [comment]
Ok. to remove test.py and keep chembl*.mm
